gcloud-java-core/src/main/java/com/google/gcloud/Restorable.java

```diff
@@ -0,0 +1,15 @@
+package com.google.gcloud;
+
+/**
+ * Implementation of this interface can persist their state and restore from it.
+ */
+public interface Restorable<T extends Restorable<T>> {
```

Comment:
I added some usage examples in this class. 
A suggestion for a better location or content are welcomed!